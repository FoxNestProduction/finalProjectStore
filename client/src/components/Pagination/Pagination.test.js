import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import AppPagination from './Pagination';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

const mockDispatch = jest.spyOn(require('react-redux'), 'useDispatch');

describe('AppPagination Component', () => {
  test('should renders AppPagination component', () => {
    const dispatch = jest.fn();
    mockDispatch.mockReturnValueOnce(dispatch);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(1);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce('');

    const { asFragment } = render(
      <AppPagination pageQty={5} />,
    );

    expect(asFragment()).toMatchSnapshot();
    const nextPage = screen.getByLabelText('Go to page 2');
    fireEvent.click(nextPage);
    expect(dispatch).toHaveBeenCalledTimes(1);
  });
});
