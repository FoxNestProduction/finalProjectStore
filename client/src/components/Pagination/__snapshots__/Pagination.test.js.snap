// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppPagination Component should renders AppPagination component 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-5zv4aj"
  >
    <nav
      aria-label="pagination navigation"
      class="MuiPagination-root MuiPagination-text css-huyi1t-MuiPagination-root"
    >
      <ul
        class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
      >
        <li>
          <button
            aria-label="Go to previous page"
            class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSecondary Mui-disabled MuiPaginationItem-previousNext css-19micn4-MuiButtonBase-root-MuiPaginationItem-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
              data-testid="NavigateBeforeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            aria-current="true"
            aria-label="page 1"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSecondary Mui-selected MuiPaginationItem-page css-19micn4-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            1
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 2"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSecondary MuiPaginationItem-page css-19micn4-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            2
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 3"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSecondary MuiPaginationItem-page css-19micn4-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            3
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 4"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSecondary MuiPaginationItem-page css-19micn4-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            4
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 5"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSecondary MuiPaginationItem-page css-19micn4-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            5
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to next page"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSecondary MuiPaginationItem-previousNext css-19micn4-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
